### PR TITLE
Update resources-repositories-repository.md

### PR DESCRIPTION
**Pull request by Matthew0x**
Hi,

- the description of "repository" keyword is not true, it's not an ID, but simply an alias.
- the "ref" keyword does not accept variables and it's the default full-path name of the branch that gets checkout by default, whenever the trigger specified in the resource object fire (so unless changed, the ref ALWAYS checks out the main/master, which may make no sense and slow down the pipeline if somebody wants to check out something else but doesn't know it)
- the triggers supposedly can use some variables, or at least wildcards like .../* or .../**/*, while other syntax structures not
- the variables are not supported in names, so the "namespace" that Azure has. Aka organization/project cannot be given as a pipeline variable, else it searches for the variable names instead of values...

I spent whole 2 days diagnosing this and it drove me mad, because it's so poorly described. I had to learn those things by trial and errors.

### CODE REVIEW

**Comment by steved0x** [comment]
Docs Build status updates of commit _[809e75c](https://github.com/Matthew0x/azure-devops-yaml-schema/commits/809e75c6d7a1c86cfad7372913ce03ef7d20a52b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[content/resources-repositories-repository.md](https://github.com/Matthew0x/azure-devops-yaml-schema/blob/patch-2/content/resources-repositories-repository.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/49a3200c-f283-8dd5-6eb3-056af594a9a5/202209090943114466-35/BuildReport?accessString=e9cb22d4af8065bceecb00223af5dccccffadc417485a07a036365adbb30c786).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/49a3200c-f283-8dd5-6eb3-056af594a9a5?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Comment by steved0x** [comment]
@Matthew0x Please accept the license from the link above and I can merge this PR. Thank you for making these additions.

**Comment by steved0x** [comment]
@Matthew0x Thank you for making this PR. We just published an update to our YAML schema reference. The new format for the YAML schema is published. The YAML schema reference now combines autogenerated portions, combined with user-editable content that can be updated by community contributors (see the [README.MD](https://github.com/MicrosoftDocs/azure-devops-yaml-schema/blob/main/README.md) in this repo for details(. The look and feel of the YAML Schema pages is the same, but behind the scenes the markdown down format is different. Unfortunately that led to a big merge conflict with your PR that I was unable to resolve. However, all of your changes have made it to the updated repository resource article which will be live momentarily.

**Comment by Matthew0x** [comment]
Thank you for trying to keep the documentation up to date. I simply worked on a specific tool that would utilize those YAML tasks and task properties and noticed that some specific things didn't work as (I presume) were theoretically expected to work. I actually had no idea how the articles work, I presumed how they worked by looking at the code.

I mitigated some of those YAML problems using a manual checkout with parameters (it seems to be supported that way, although still broken. I might want to open a ticket, that's unrelated to this case; the repository type of resource references).

This specific change wasn't very important, although I think that  some people who like to parametrize their YAML pipelines might bang their heads against the wall because of that. Especially due to the triggers, which weirdly enough support the wildcards, but not the variables (I think it's still true, however I haven't tested that specific scenario in a while). 

Summing up - I will read the new rules/learn the new syntax in case I can ever add something to the DOCs. 
Thank you a lot :+1:
